hkmc2/shared/src/main/scala/hkmc2/codegen/Lifter.scala

```diff
@@ -824,145 +882,243 @@ class Lifter(handlerPaths: Opt[HandlerPaths])(using State, Raise):
           val mainDefn = FunDefn(f.owner, f.sym, PlainParamList(extraParamsCpy) :: headPlistCopy :: Nil, bdy)
           val auxDefn = FunDefn(N, singleCallBms, flatPlist, lifted.body)
           
-          if ctx.firstClsFns.contains(f.sym) then
-            Lifted(mainDefn, auxDefn :: extras)
-          else
-            Lifted(auxDefn, extras) // we can just include the flattened defn
-            
-        case c: ClsLikeDefn if !modOrObj(c) =>
-          val newDef = c.copy(
-            owner = N, auxParams = c.auxParams.appended(PlainParamList(extraParams))
-          )
-          val Lifted(lifted, extras) = liftDefnsInCls(newDef, newCtx)
-
-          val bms = fakeCtorBms.get
-
-          // create the fake ctor here
-          inline def mapParams(ps: ParamList) = ps.params.map(p => VarSymbol(p.sym.id))
 
-          val paramSyms = c.paramsOpt.map(mapParams)
-          val auxSyms = c.auxParams.map(mapParams)
-          val extraSyms = extraParams.map(p => VarSymbol(p.sym.id))
-
-          val paramArgs = paramSyms.getOrElse(Nil).map(_.asPath)
-
-          inline def toPaths(l: List[Local]) = l.map(_.asPath)
+          Lifted(mainDefn, auxDefn :: extras)
+        case c: ClsLikeDefn =>
+          val createSym: String => (VarSymbol, LocalPath.PubField) = 
```

Comment:
This should probably be a `def`. You can move the `fresh` definition to the line above.